crawler.js

```diff
@@ -505,6 +540,63 @@ class Crawler {
     // extra wait for all resources to land into WARCs
     console.log("Waiting 5s to ensure WARCs are finished");
     await this.sleep(5000);
+    
+    if (this.params.combineWARC) {
+      console.log("Combining the warcs");
+      
+  
+      // Get the list of created Warcs
+      const warcLists = fs.readdirSync(path.join(this.collDir, "archive"));
+  
+      const fileSizeObjects = []; // Used to sort the created warc by fileSize 
+      
+      // Used to name the combined warcs
+      var combinedWarcNumber = 0;
+      var combinedWarcName = this.params.collection.concat("_combined_", combinedWarcNumber.toString(),".warc");
+      
+      // Create the header for the first combined warc
+      const warcBuffer = await this.createWARCInfo(combinedWarcName);
+
+      // Go through a list of the created works and create an array sorted by their filesize with the largest file first.
+      for (var i = 0; i < warcLists.length; i++) {
+        var fileName = path.join(this.collDir, "archive", warcLists[i])
+        var fileSize = await this.getFileSize(fileName)
+        fileSizeObjects.push({'fileSize': fileSize, 'fileName': fileName});
+        fileSizeObjects.sort(function(a, b){
+            return b.fileSize - a.fileSize;
+        });
+      }
+      
+      // Write out the header for the first combined warc file
+      fs.writeFileSync(path.join(this.collDir, "archive", combinedWarcName), warcBuffer);
+      var generatedCombinedWarcs = []
+      generatedCombinedWarcs.push(combinedWarcName)
+      
+      // Iterate through the sorted file size array. 
+      for (var j = 0; i < fileSizeObjects.length; j++){
+        // Check the size of the existing combined warc.
+        var currentCombinedWarcSize = await this.getFileSize(path.join(this.collDir, "archive", combinedWarcName));
+        //  If adding the current warc to the existing combined file creates a file smaller than the rollover size add the data to the combinedWarc
+        var proposedWarcSize = fileSizeObjects[j].fileSize; + currentCombinedWarcSize;
+        if (proposedWarcSize < this.params.rolloverSize){
+          fs.appendFileSync(path.join(this.collDir, "archive", combinedWarcName), fs.readFileSync(fileSizeObjects[j].fileName));
+        }
+        // If adding the current warc to the existing combined file creates a file larger than the rollover size do the following: 
+        // 1. increment the combinedWarcNumber 
+        // 2. create the name of the new combinedWarcFile
+        // 3. Write the header out to the new file 
+        // 4. Write out the current warc data to the combinedFile
+        else{
+          combinedWarcNumber = combinedWarcNumber + 1;
+          const combinedWarcName = this.params.collection.concat("_", combinedWarcNumber.toString(),".warc");
```

Comment:
this can just be the below, with template-literal style:
```suggestion
          const combinedWarcName =`${this.params.collection}_${combinedWarcNumber}.warc`;
```